geopyspark/geotrellis/layer.py

```diff
@@ -255,12 +255,63 @@ def to_tiled_layer(self, extent=None, layout=None, crs=None, tile_size=256,
             ``extent`` and ``layout`` must both be defined if they are to be used.
 
         Returns:
-            :class:`~geopyspark.geotrellis.rdd.TiledRasterLayer`
+            :class:`~geopyspark.geotrellis.rdd.RasterLayer`
         """
 
         return self.tile_to_layout(self.collect_metadata(extent, layout, crs, tile_size),
                                    resample_method)
 
+    def bands(self, band):
+        """Select a subsection of bands from the ``Tile``\s within the layer.
+
+        Note:
+            There could be potential high performance cost if operations are performed
+            between two sub-bands of a large data set.
+
+        Note:
+            Due to the natue of GeoPySpark's backend, if selecting a band that is out of bounds
+            then the error returned will be a ``py4j.protocol.Py4JJavaError`` and not a normal
+            Python error.
```

Comment:
Drat, thats too much potential confusion to leave hanging around. We could return `Either[RasterLayer[K], Exception]` or `Either[RasterLayer[K], String]`. 

Both `Left` and `Right` have `.value` so python caller could call `result.value` and match on the type. Since the call will go through reflection, we know we will get something back.
Maybe this would even warrant making a py4j decoder to go from scala `Exception` to python error.